packages/frontend/src/components/Explorer/ExploreTree/TableTree/Tree/TreeSingleNode.tsx

```diff
@@ -10,23 +10,128 @@ import {
     Item,
     timeFrameConfigs,
 } from '@lightdash/common';
-import { Group, Highlight, NavLink, Text, Tooltip } from '@mantine/core';
+import {
+    Anchor,
+    Flex,
+    Group,
+    Highlight,
+    HoverCard,
+    NavLink,
+    Text,
+    Title,
+    Tooltip,
+    useMantineTheme,
+} from '@mantine/core';
+import { IconAlertTriangle, IconFilter } from '@tabler/icons-react';
+import MarkdownPreview from '@uiw/react-markdown-preview';
 import { darken, lighten } from 'polished';
 import { FC } from 'react';
 import { useToggle } from 'react-use';
-
-import { IconAlertTriangle, IconFilter } from '@tabler/icons-react';
 import { getItemBgColor } from '../../../../../hooks/useColumns';
 import { useFilters } from '../../../../../hooks/useFilters';
 import FieldIcon from '../../../../common/Filters/FieldIcon';
 import MantineIcon from '../../../../common/MantineIcon';
+import { useItemDetail } from '../ItemDetailContext';
 import { Node, useTableTreeContext } from './TreeProvider';
 import TreeSingleNodeActions from './TreeSingleNodeActions';
 
 type Props = {
     node: Node;
 };
 
+/**
+ * Truncate an item description by limiting to:
+ *
+ * - No more than 5 lines
+ * - No more than 256 characters total (BEFORE markdown rendering)
+ * - Breaks before any code blocks after the first line
+ * - Breaks before anything that looks like a table after the first line
+ */
+const truncateDescription = (description: string) => {
+    const lines = description.split('\n');
+
+    /**
+     * Collect up to 5 description lines, breaking early if we find anything that
+     * looks like a code block or a table.
+     */
+    const truncatedLines = lines
+        .slice(0, 4)
+        .reduce<string[]>((truncated, line, idx) => {
+            if (idx === 0 || (line.at(0) !== '|' && !line.startsWith('```'))) {
```

Comment:
so if the description starts with a block of code we will not show anything, right ?  I guess it kind of makes sense...